### ISSUE TITLE
obscure error when module to `protect` does not exist

### ISSUE THREAD

**Issue by msz** [msz/hammox#22]
```
** (MatchError) no match of right hand side value: :error
     stacktrace:
       (hammox) lib/hammox.ex:336: Hammox.fetch_typespecs/3
       (hammox) lib/hammox.ex:322: Hammox.fetch_typespecs!/3
       (hammox) lib/hammox.ex:121: Hammox.protect/2
       (hammox) lib/hammox.ex:178: anonymous fn/4 in Hammox.protect/3
       (elixir) lib/enum.ex:1336: Enum."-map/2-lists^map/1-0-"/2
       (elixir) lib/enum.ex:2994: Enum.flat_map_list/2
       (hammox) lib/hammox.ex:168: Hammox.protect/3
```

Should show a good explanation what happened instead.

**Comment by msz** [comment]
Spotted when the behaviour module name was invalid but might break with wrong implementation module too.

**Comment by msz** [comment]
Closed in #52
